applications/nrf_desktop/doc/ble_state.rst

```diff
@@ -3,4 +3,82 @@
+|                |                         |               |                       +-----------------------+
+|                |                         |               |                       | :ref:`ble_discovery`  |
+|                |                         |               |                       +-----------------------+
+|                |                         |               |                       | :ref:`hid_state`      |
+|                |                         |               |                       +-----------------------+
+|                |                         |               |                       | :ref:`hid_forward`    |
+|                |                         |               |                       +-----------------------+
+|                |                         |               |                       | ``ble_state``         |
++----------------+-------------------------+---------------+-----------------------+-----------------------+
+
+Configuration
+*************
+
+The module requires only basic Bluetooth configuration (described in the Bluetooth guide).
+
+The option ``CONFIG_DESKTOP_BLE_ENABLE_PASSKEY`` can be used to enable passkey based pairing for increased security.
+The :ref:`passkey` module has to be enabled and configured in that case.
+
+Implementation details
+**********************
+
+The ``ble_state`` module is used both by Bluetooth Periperal and Central devices.
+
+The module propagates information about the connection state changes using ``ble_peer_event``.
+:cpp:member:`id` is a pointer to the connection object.
+:cpp:member:`state` is the connection state.
+
+Connection state can be set to one of the following values:
+
+* :cpp:enum:`PEER_STATE_CONNECTED` - remote peer successfully connected.
+* :cpp:enum:`PEER_STATE_CONN_FAILED` - failed to connect remote peer.
+* :cpp:enum:`PEER_STATE_SECURED` - security for the connection set at least to level 2 (encryption and no authentication).
+* :cpp:enum:`PEER_STATE_DISCONNECTED` - remote peer disconnected.
+
+For Bluetooth Peripheral, the ``ble_state`` module is used to request connection security level 2.
+If the connection security level 2 is not successfully established, the Peripheral device disconnects.
+
+In case ``CONFIG_DESKTOP_BLE_ENABLE_PASSKEY`` option is set, ``ble_state`` module registers set of authenticated pairing callbacks (:c:type:`struct bt_conn_auth_cb`).
+The callbacks can be used to achieve higher security levels.
+The passkey input is handled in the :ref:`passkey` module.
+
+If Nordic proprietary BLE Link Layer is selected (:option:`CONFIG_BT_LL_NRFXLIB`), the module sends Bluetooth HCI command to enable the LLPM when Bluetooth is ready.
+For the mentioned Link Layer, the ``ble_state`` module is also used to set TX power for connections.
+
+``ble_state`` module keeps references to :c:type:`struct bt_conn` objects to ensure that they remain valid when other application modules access them.
```

Comment:
Can we move lines 82-84 before line 75, that is before discussing the optional passkey and BLE LL topics?

Reply:
Will do